config/manager/controller-manager.yaml

```diff
@@ -49,7 +49,40 @@ spec:
       # versions < 1.19 or on vendors versions which do NOT support this field by default (i.e. Openshift < 4.11 ).
       # seccompProfile:
       #   type: RuntimeDefault
-      initContainers: []
+      initContainers:
+        
+        # temporary init container to patch all isbsvc on the cluster owned by ISBServiceRollout to add "promoted" label
+        - name: label-isbsvc
+          image: bitnami/kubectl
+          command: 
+
+            - "/bin/bash"
+            - "-c"
+            - |
+              #!/bin/bash
+
+              # Get all namespaces
+              namespaces=$(kubectl get namespaces -o jsonpath='{.items[*].metadata.name}')
+
+              # Loop through each namespace
+              for namespace in $namespaces; do
+                echo "Processing namespace: $namespace"
+                
+                # Get all isbsvc in the current namespace
+                interstepbuffersvcs=$(kubectl get isbsvc -n $namespace -o json | jq -r '.items[] | select(.metadata.ownerReferences[]?.kind == "ISBServiceRollout") | .metadata.name')
+                
+                # Loop through each isbsvc
+                for isbsvc in $interstepbuffersvcs; do
+                  echo "Patching isbsvc: $isbsvc in namespace: $namespace"
+                  
+                  # Patch the isbsvc to add the labels
+                  kubectl patch isbsvc $isbsvc -n $namespace --type='merge' -p "{\"metadata\": {\"labels\": {\"numaplane.numaproj.io/upgrade-state\": \"promoted\"}}}"
+                  kubectl patch isbsvc $isbsvc -n $namespace --type='merge' -p "{\"metadata\": {\"labels\": {\"numaplane.numaproj.io/parent-rollout-name\": \"$isbsvc\"}}}"
+                done
+              done
+
+              echo "All interstepbufferservices have been patched."
+
```

Comment:
Is this patch something that will be executed while transitioning between Numaplane versions or something permanent?

Reply:
temporary while transitioning between numaplane versions

Reply:
I can create an issue so we don't forget to remove it

Reply:
or you can add a TODO comment if preferred

Reply:
I'll do both